examples/main/main.cpp

```diff
@@ -147,22 +150,20 @@ int main(int argc, char ** argv) {
     }
 
     // number of tokens to keep when resetting context
-    if (params.n_keep < 0 || params.n_keep > (int)embd_inp.size() || params.instruct) {
+    if (params.n_keep < 0 || params.n_keep > (int)embd_inp.size()) {
```

Comment:
FYI, user now should explicitly call `--keep -1`. This is added to configs, so it should not obstacle for new users.